flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/calls/ScalarFunctionCallGen.scala

```diff
@@ -44,14 +44,24 @@ class ScalarFunctionCallGen(
       operands: Seq[GeneratedExpression])
     : GeneratedExpression = {
     // determine function signature and result class
-    val matchingSignature = getSignature(scalarFunction, signature)
+    val matchingMethod = getEvalMethod(scalarFunction, signature)
       .getOrElse(throw new CodeGenException("No matching signature found."))
+    val matchingSignature = matchingMethod.getParameterTypes
     val resultClass = getResultTypeClass(scalarFunction, matchingSignature)
 
+    // zip for variable signatures
+    var paramToOperands = matchingSignature.zip(operands)
+    var i = paramToOperands.length
+    while (i < operands.length
```

Comment:
Forget zipAll, i think you can write codes like this instead of a while loop
```
    if (operands.length > matchingSignature.length) {
      operands.drop(matchingSignature.length).foreach(op =>
        paramToOperands = paramToOperands :+ (matchingSignature.last.getComponentType, op))
    }
```